tasks.py

```diff
@@ -801,8 +814,24 @@ def tests(context, failfast=False, keepdb=False, lint_only=False):
     pylint(context)
     print("Running mkdocs...")
     build_and_check_docs(context)
+    print("Checking app config schema...")
+    validate_app_config(context)
     if not lint_only:
         print("Running unit tests...")
         unittest(context, failfast=failfast, keepdb=keepdb)
         unittest_coverage(context)
     print("All tests have passed!")
+
+
+@task
+def generate_app_config_schema(context):
+    """Generate the app config schema based on the app config."""
+    start(context, service="nautobot")
+    nbshell(context, file="development/app_config_schema.py", env={"APP_CONFIG_SCHEMA_COMMAND": "generate"})
+
+
+@task
+def validate_app_config(context):
+    """Validate the app config based on the app config schema."""
+    start(context, service="nautobot")
+    nbshell(context, plain=True, file="development/app_config_schema.py", env={"APP_CONFIG_SCHEMA_COMMAND": "validate"})
```

Comment:
To import the App config class, it's necessary to do it in initialized Nautobot / Django environment. It is possible to use AST parser instead, but this seems to be a more simple solution.

Another step is to deprecate `default_settings` and `required_settings` config class properties, and replace it with JSON schema. For Nautobot 2.x, it can work the way, that when the schema file is present, it will be used and deprecation message will be logged out. In Nautobot 3.x, we can remove these properties.

Another improvement I suggest, is to store the App configuration as a property of it's config class, e.g. `NautobotAppConfig.settings`. All apps code then can use this, instead of accessing `django.PLUGINS_CONFIG`.

WDYT @whitej6 @lampwins ?